R/tidy_output.R

```diff
@@ -0,0 +1,318 @@
+# Copyright 2022 Bedford Freeman & Worth Pub Grp LLC DBA Macmillan Learning.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# .make_token_map ---------------------------------------------------------
+
+
+#' Make Token Map
+#'
+#' Given a list of input token sequences, construct a data frame with explicit
+#' index columns. Later this can be joined onto the indexed outputs to add the
+#' actual token and segment columns to the tidy data frames.
+#'
+#' @inheritParams tidy_bert_output
+#'
+#' @return A data frame of the input tokens, with explicit index columns.
+#' @keywords internal
+.make_token_map <- function(tokenized) {
+  tok_id_list <- tokenized$token_ids
+  tok_list <- purrr::map(tok_id_list, names)
+  tt_list <- tokenized$token_type_ids
+
+  token_map <- tibble::tibble(
+    sequence_index = integer(),
+    token_index = integer(),
+    segment_index = integer(),
+    token = character()
+  )
+  for (i in seq_along(tok_list)) {
+    tokens <- tok_list[[i]]
+    token_ids <- tok_id_list[[i]]
+    segments <- tt_list[[i]]
+    token_seq <- seq_along(tokens)
+    token_map <- dplyr::bind_rows(
```

Comment:
I'm pretty sure it's generally faster to generate a list of df's in the `for` and then `bind_rows` at the end (the argument to `bind_rows` can be a list of dfs). I'm not going to request that change 'cuz this works and will change quite a bit once we update {tt}, but FYI.

Reply:
good call!